rust_src/src/strings.rs

```diff
@@ -202,10 +202,9 @@ fn string_to_unibyte(string: LispObject) -> LispObject {
     }
 }
 
-// @TODO need to rework this function to use new as_symbol_or_error API
 fn get_string_or_symbol(mut string: LispObject) -> multibyte::LispStringRef {
     if string.is_symbol() {
-        string = string.symbol_name()
+        string = string.as_symbol_or_error().symbol_name()
```

Comment:
This still duplicates the symbol check. I'd make it
```
match string.as_symbol() {
    Some(sym) => sym.symbol_name().as_string().expect("symbol name not a string?")
    None => string.as_string_or_error()
}
```
If you like, you can also make `string_equal` use this function, it currently uses SYMBOL_NAME directly.